pep/001-initial-pep-process.md

```diff
@@ -0,0 +1,386 @@
+decisions were made.  It should describe alternate designs that were considered
+and related work.
+
+The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during discussion.
+
+
+#### Specification
+
+The technical specification should describe the command and/or configuration
+syntax and semantics of any new feature.
+
+* If this is a command, we are looking for what the `porter COMMAND**: help`
+  would look like: description of command, arguments, flags, default behavior
+  and error handling.
+
+* If this is a syntax change to a configuration file, define the allowed syntax,
+  at least one example per use case, covering defaults and error handling.
+
+* All PEPs will be reviewed for user experience. So make sure to think about the
+  common use case, how people can accomplish more advanced scenarios, precedence
+  from existing Porter features or other tools in the ecosystem, and how the
+  change fits into Porter workflows and tasks.
+
+
+#### Backwards Compatibility
+
+All PEPs that introduce backwards incompatibilities must include a section
+describing these incompatibilities and their severity.  The PEP must explain how
+to deal with these incompatibilities, possibly with defaulting or migrations.
+PEP submissions without a sufficient backwards compatibility treatise may be
+rejected outright.
+
+
+#### Security Implications
+
+If there are security concerns in relation to the PEP, those concerns should be
+explicitly written out to make sure reviewers of the PEP are aware of them.
+
+
+#### Rejected Ideas
+
+Throughout the discussion of a PEP, various ideas will be proposed which are not
+accepted. Those rejected ideas should be recorded along with the reasoning as to
+why they were rejected. This both helps record the thought process behind the
+final version of the PEP as well as preventing people from bringing up the same
+rejected idea again in subsequent discussions.
+
+
+#### Open Issues
+
+While a PEP is in draft, ideas can come up which warrant further discussion.
+Those ideas should be recorded so people know that they are being thought about
+but do not have a concrete resolution. This helps make sure all issues required
+for the PEP to be ready for consideration are complete complete and reduces
+people duplicating prior discussion.
+
+
+### PEP Lifecycle
+
+We are using the process from Kubernetes Enhancement Proposal for our status's
+because they work better for our implementation than Python (Porter isn't a spec
+like python, so Kubernetes process fits us better).
+
+This is the happy path, where a PEP is accepted.
+
+1. The idea is posted to the Porter [forum] where the maintainers and community
+   can see it and comment. A maintainer will let you know if it is suitable for
+   a PEP and assign it a number.
+
+1. A pull request for the PEP is submitted with an initial status of
```

Comment:
Before we merge the PR that status is **proposed**. We may not merge it and it could go nowhere. Once it is merged the status is **provisional**. 

I didn't want to have them submit it as proposed in the md, and then after merging have to go back and update the status to provisional. So I thought open pull requests for a new PEP are proposed, and it's not provisional until it's merged. Let me know if you have suggestions to make that less confusing!

Reply:
I think a drawing at the top of this to show states would be super helpful. The states make sense to me but it can be challenging to explain with words.